### PR TITLE
Server side handlebars rendering and fix interceptor chain

### PR DESCRIPTION
**Pull request by ricksbrown**
This contains two main items:

1. Backport of the interceptor chain fix
2. Implement server side handlebars rendering (which obviously uses the interceptor chain)

This is primarily about making IE8 less broken by moving these (related) steps from the browser to the server:

- Handlebars rendering
- Theme i18n

Edit: also tweak to dismiss loading shim which appears to "hang" on IE8.

### CODE REVIEW

**Comment by ricksbrown** [comment]
@jonathanaustin PTAL

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/template/HandlebarsRendererImpl.java [hunk 1]
```diff
@@ -54,6 +56,16 @@
...
 
 	/**
+	 * Perform theme i18n on the server, defaults is "true" (theme i18n bundle must be on the classpath).
+	 */
+	public static final String THEME_I18N = "THEME_I18N";
+
+	/**
+	 * The basename of the theme i18n resource bundle. (TODO should this be a configuration property?)
+	 */
+	private static final String THEME_I18N_RESOURCE_BUNDLE_BASE_NAME = "com/github/bordertech/wcomponents/theme-messages";
```
Comment (jonathanaustin) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/template/HandlebarsRendererImpl.java]: Noticed you half setup the parameter name for this in the `ConfigurationProperties`. Might as well set it up with this PR.

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/ConfigurationProperties.java [hunk 2]
```diff
@@ -166,6 +166,11 @@
 	public static final String I18N_RESOURCE_BUNDLE_BASE_NAME = "bordertech.wcomponents.i18n.baseName";
 
 	/**
+	 * The resource bundle base name.
+	 */
+	public static final String I18N_THEME_RESOURCE_BUNDLE_BASE_NAME = "bordertech.wcomponents.i18n.baseName";
```
Comment (jonathanaustin) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/ConfigurationProperties.java]: Might as well create the getter and default for this.

**Comment by jonathanaustin** [comment]
Made two comments. Besides those it LGTM.

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/template/HandlebarsRendererImpl.java (hunk 1 above)
Comment (ricksbrown) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/template/HandlebarsRendererImpl.java]: Doh yeah, good catch

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/ConfigurationProperties.java (hunk 2 above)
Comment (ricksbrown) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/ConfigurationProperties.java]: yep

**Pull request merged**